lib/slurm_squeue_client.rb

```diff
@@ -61,7 +61,73 @@ def sinfo
     
     s.success? ? @sinfo = o : raise(CommandFailed, e)
   end
-  
+
+  # Return length of GRES field from SLURM
+  # @return [Integer] gres length
+  def gres_length
+    return @gres_length if defined?(@gres_length)
+
+    Open3.pipeline_rw "sinfo -o '%G' | awk '{ print length }'", "sort -n", "tail -1" do |stdin,stdout|
+      stdin.write stdout
+      stdin.close
```

Comment:
In this case it seems like you would want to use `Open3.pipeline_r` instead of `Open3.pipeline_rw` as you don't need to write anything to stdin of the sinfo command, you just need to read stdout.